
This patch fixes a Oops in arp_rcv() when proxy-arp is used

--- linux/net/ipv4/arp.c.orig	Thu Jun 28 17:29:10 2001
+++ linux/net/ipv4/arp.c	Tue Jul  3 19:37:25 2001
@@ -738,7 +738,7 @@
 			    (addr_type == RTN_UNICAST  && rt->u.dst.dev != dev &&
 			     (IN_DEV_PROXY_ARP(in_dev) || pneigh_lookup(&arp_tbl, &tip, dev, 0)))) {
 				n = neigh_event_ns(&arp_tbl, sha, &sip, dev);
-				neigh_release(n);
+				if (n) neigh_release(n);
 
 				if (skb->stamp.tv_sec == 0 ||
 				    skb->pkt_type == PACKET_HOST ||
